### PR TITLE
fix ISSUE-361: Pulsar manager's swagger dose not work.

### PR DESCRIPTION
**Pull request by yixing1337**
Fixes #ISSUE-361

### Motivation
Pulsar manager's swagger does not work, so I try to fix this issue.

### Modifications
1. Swagger urls can't be accessed because the interceptors  blocked them, so I modified the interceptors to allow swagger urls.
2. I added a swagger configration in `application.properties`, thus, if you want to use swagger in development environment, you can set the value `true`. And  if you don't want to enable swagger in production environment, you can set the value `false`.
3. I modified the `SwaggerConfig` class, because swagger need some authorization infomation, otherwise it will receive 401 code in response header.
4. I modified the README.md to show how to use swagger, and added some images in docs.
5. Now the swagger can work correctly.

### Verifying this change

- [x] Make sure that the change passes the `./gradlew build` checks.

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -187,6 +187,42 @@ Set the `bkvm.enabled` field to `true` in the file [bkvm.conf](https://github.co
         apachepulsar/pulsar-manager:v0.2.0
     ```
 
+### Enable swagger ui
+If you want to enable swagger ui, please set the following option to true in file `application.properties`, default value is true.
+if you don't want to enable swagger ui in production environment, you can set the value false.
+
+```$xslt
+swagger.enabled=true
+```
```
Comment (Jennifer88huang) [inline src/README.md:196]:
```suggestion
### Enable swagger UI
To enable swagger UI, set `swagger.enabled=true` in the `application.properties` file. The default value is `true`.

If you want to disable swagger UI in the  production environment, you can set `swagger.enabled=false`.

File: src/README.md [hunk 2]
```diff
@@ -187,6 +187,42 @@ Set the `bkvm.enabled` field to `true` in the file [bkvm.conf](https://github.co
...
+
+4. Click the `Authorize` button in swagger ui, and then fill in the authorize information of swagger
+     - environment: the environment name you configed in pulsar manager
+     - token: the token you copied above
+     - username: the username you just login
+    
+    ![swagger-authorize](../docs/img/swagger-authorize.png)
+
+5. Then you can directly request other apis and receive correct response, because it has been authorized.
+
```
Comment (Jennifer88huang) [inline src/README.md:224]:
```suggestion
### How to use swagger to access the API
1. Enable swagger UI by setting 'swagger.enabled=true' in the `application.properties` file.

2. Visit http://YOUR_BACKEND_SERVICE_HOST:7750/swagger-ui.html

3. Request the login API, use the username and password of the Pulsar Manager to login, copy the `token` of the response header.
   
    The request body is as follows:
    ```$xslt
    {
      "username": "admin",
      "password": "apachepulsar"
    }
    ```
    Copy the token of the response header.
    swagger-request-login

4. Authorize. Click the `Authorize` button in swagger UI, and then fill in the authorization information of swagger.
     - environment: the environment name you have configured in Pulsar Manager.
     - token: the token you have copied in the previous step.
     - username: the username you login.
    
    swagger-authorize

5. Request other APIs.
```

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -153,3 +153,6 @@ tls.pulsar.admin.ca-certs=ca-client-path
 
 # support peek message, default false
 pulsar.peek.message=false
+
+# swagger configration
+swagger.enabled=true
```
Comment (tuteng) [inline src/main/resources/application.properties:158]: I suggest we disabled it by default, many users will deploy it directly online, which may bring some security issues, if users want to enable it on, they can configure it themselves

Comment (yixing1337) [reply src/main/resources/application.properties]: yes, good suggestion, I set this property false, and commit again

**Review by tuteng — APPROVED** [review]
LGTM

**Review by Jennifer88huang — APPROVED** [review]

**Comment by abenbouz** [comment]
hello 
i have tested last release 0.2.1 but i have an issue with pulsar manager swagger ui 
when i access the url  http://backend:7750/swagger-ui.html  i have the flowing error  : 

{"message":"Please login."}

have any idea on how to solve that ?
Thanks

**Comment by yixing1337** [comment]
@abenbouz
Hi, I have tried again, and make sure the swagger works well.
Did you config the `swagger.enabled=true` in application.properties?
And then you can follow the steps of `Enable swagger UI` in pulsar-manager/src/README.md.

**Comment by abenbouz** [comment]
@ yixing1337  
thanks for your answer 

i have updated the application properties  :  swagger.enabled=true

and then for the Readme i have the html message in the second step 

2 - Visit http://YOUR_BACKEND_SERVICE_HOST:7750/swagger-ui.html  --> 
 {"message":"Please login."}

**Comment by yixing1337** [comment]
@abenbouz 
Hi, You mentioned earlier that the release version you downloaded is v0.2.1, but the latest version is v0.2.0
the v0.2.0 was released on 2020-09-28, but this pr was merged later on 2020-12-29.
so if you want to use the swagger feature, you have to compile from the master branch.
and what's problems did you have when you compile from the source? maybe I can help you to fix it.   :)

**Comment by abenbouz** [comment]
hi ,

yes version 0.2.1 from stream native https://github.com/streamnative/pulsar-manager/releases/tag/v0.2.1

this version includes your fix 

for the compile it's an internal problem linked to internet access

**Pull request merged**